constraint-solver/tests/solver.rs

```diff
@@ -98,3 +114,126 @@ fn bit_decomposition() {
         ],
     );
 }
+
+const BYTE_BUS_ID: u64 = 42;
+const XOR_BUS_ID: u64 = 43;
+
+struct TestBusInteractionHandler {}
+impl BusInteractionHandler for TestBusInteractionHandler {
+    type T = GoldilocksField;
+    type V = Var;
+
+    fn handle_bus_interaction(
+        &self,
+        bus_interaction: BusInteraction<RangeConstraint<GoldilocksField>>,
+    ) -> BusInteraction<RangeConstraint<GoldilocksField>> {
+        let (Some(bus_id), Some(multiplicity)) = (
+            bus_interaction.bus_id.try_to_single_value(),
+            bus_interaction.multiplicity.try_to_single_value(),
+        ) else {
+            return bus_interaction;
+        };
+
+        if multiplicity.is_zero() {
+            return bus_interaction;
+        }
+
+        assert!(multiplicity.is_one(), "Only expected send interactions");
+        let byte_constraint = RangeConstraint::from_mask(0xffu32);
+        let payload_constraints = match bus_id.to_degree() {
+            BYTE_BUS_ID => {
+                assert_eq!(bus_interaction.payload.len(), 1);
+                vec![byte_constraint]
+            }
+            XOR_BUS_ID => {
+                assert_eq!(bus_interaction.payload.len(), 3);
+                if let (Some(a), Some(b)) = (
```

Comment:
If we use that for a the actual implementation in openvm, we should consider all three cases where we know two values and can compute the third.